docs/src/reference/configuration.rst

```diff
@@ -3,62 +3,89 @@ Configuration parameters for Nextstrain SARS-CoV-2 workflow
 
 .. contents:: Table of Contents
    :local:
-   :depth: 1
+   :depth: 2
 
-S3_DST_BUCKET
--------------
+Main Workflow configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
--  type: string
--  description: S3 bucket to store files from the ``upload`` rule in ``export_for_nextstrain.smk``. Currently only available to Nextstrain builds.
+Parameters in this section define the main inputs and outputs of the workflow, as well as the important ``subsampling`` parameter. Often these will be the only parameters you need to modify.
 
-S3_DST_COMPRESSION
-------------------
+inputs
+------
 
--  type: string
--  description: Compression format to use for files uploaded to S3 by the ``upload`` rule.
--  examples
+-  type: array
+-  description: A list of named input datasets to use for the workflow. Input order determines the precedence of genome sequences and metadata such that later datasets override earlier datasets. Each input must define a ``name``, a path to ``metadata``, and a path to sequences at one of many possible starting points. The workflow merged all input metadata and sequences into a single metadata and sequences file prior to subsampling.
+-  required
 
-   -  ``xz``
-   -  ``gz``
+   -  ``name``
+   -  ``metadata``
+   -  ``sequences`` or ``aligned``
 
-S3_DST_ORIGINS
---------------
+-  examples:
 
--  type: array
--  items:
+.. code:: yaml
 
-   -  type: string
+   inputs:
+     - name: example-data
+       metadata: data/example_metadata.tsv.xz
+       sequences: data/example_sequences.fasta.xz
+     - name: prealigned-data
+       metadata: data/other_metadata.tsv.xz
+       aligned: data/other_aligned.fasta.xz
 
--  description: List of input names (i.e., “origins”) for which intermediate files should be uploaded to S3 by the ``upload`` rule.
--  examples
+Valid attributes for list entries in ``inputs`` are provided below.
 
-   -  ``["gisaid"]``
+.. contents::
+   :local:
```

Comment:
> @jameshadfield: I wasn't sure what the following directive was doing

This adds a "table of contents" for individual sub-headings. Since we are only exposing the top-level YAML keys using `:depth: 2` in the main TOC, deeper keys like `builds.title` don't have a link up top. These smaller scoped TOCs provide these links:

<img width="500" alt="builds-toc" src="https://user-images.githubusercontent.com/13424970/162284925-146e2dc6-4c2e-4cf7-a829-f838edc1df96.png">